blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/Granularity.java

```diff
@@ -254,23 +280,25 @@ private static Granularity granularityFromPointsLinear(double requestedDuration,
      *
      * @param requestedDuration (milliseconds)
      */
-    private static Granularity granularityFromPointsGeometric(String tenantid, long from, long to, double requestedDuration, int requestedPoints) {
+    private static Granularity granularityFromPointsGeometric(String tenantid, long from, long to, double requestedDuration, int requestedPoints, long assumedIntervalMillis, boolean checkTtl) {
         double minimumPositivePointRatio = Double.MAX_VALUE;
         Granularity gran = null;
-        if (SAFETY_TTL_PROVIDER == null) {
+        if (checkTtl && SAFETY_TTL_PROVIDER == null) {
             SAFETY_TTL_PROVIDER = SafetyTtlProvider.getInstance();
         }
 
```

Comment:
These code changes are not mere refactoring. They are altering the functionality. What story are these changes part of?